clients/python/templates/client.mustache

```diff
@@ -12,20 +12,37 @@ from {{apiPackage}} import {{classFilename}}
 class _WrappedApiClient(ApiClient):
     """ApiClient that fixes some weirdness"""
 
-    # Wrap files_parameters to work with unnamed "files" (e.g. MemIOs).
     def files_parameters(self, files=None):
-        if files is not None:
-            for (param_name, file_instances) in files.items():
-                i = 0
-                if file_instances is None:
-                    continue
-                for file_instance in file_instances:
-                    if file_instance is not None and not hasattr(file_instance, 'name'):
-                        # Generate a fake name.
-                        i += 1
-                        file_instance.name = f'{param_name}{i}'
-        return super().files_parameters(files)
+        """
+        Transforms input file data into a formatted list and combines it with superclass file parameters.
+        In case file_name is a string we assume that it is a path to the file that needs to be read.
+        In case file_name is bytes we assume that it is a file-like object that we append the information.
+        The parent class will handle the files to read.
```

Comment:
What's a "Superclass file parameter"?  I can figure out that it's the `file_parameters` of the base class.  But when I read it, it sounds like the secret weapon of a Bond villain.

Perhaps

```suggestion
        Transforms input file data into a formatted list to return file_parameters.
        Assume a string file_name is a path to the file to read.
        Assume a bytes file_name is a file-like object that we append the information.
        The parent class will handle the files to read.
```

But I am still not sure how "bytes" can be "a file-like object".

Reply:
> What's a "Superclass file parameter"? I can figure out that it's the `file_parameters` of the base class. But when I read it, it sounds like the secret weapon of a Bond villain.
> 
> Perhaps
> 
> But I am still not sure how "bytes" can be "a file-like object".

Same as in https://github.com/treeverse/lakeFS/pull/2244
The SDK generated code will map the post parameters which it defines as `files` and will this function passing a dictionary that holds {"content": "filename" or ["filename",...]}.
The this function process the list of filenames and transform them into a list of tuples (params) - (content, <file content>, <content type>).
The list of params will be posted like other params by the client post.